src/net_processing.cpp

```diff
@@ -1373,7 +1375,11 @@ static void RelayAddress(const CAddress& addr, bool fReachable, const CConnman&
     // Use deterministic randomness to send to the same nodes for 24 hours
     // at a time so the m_addr_knowns of the chosen nodes prevent repeats
     uint64_t hashAddr = addr.GetHash();
-    const CSipHasher hasher = connman.GetDeterministicRandomizer(RANDOMIZER_ID_ADDRESS_RELAY).Write(hashAddr << 32).Write((GetTime() + hashAddr) / (24 * 60 * 60));
+
+    std::chrono::seconds current_time = GetTime<std::chrono::seconds>();
+    // Adding address hash makes exact rotation time different per address, while preserving periodicity.
+    uint64_t time_addr = (current_time + std::chrono::hours(hashAddr)) / ROTATE_ADDR_RELAY_DEST_INTERVAL;
```

Comment:
Previously `hashAddr` was of type seconds, now it is type hours. What was the reason to change it?

Reply:
I think i changed it while trying something else. There is no real difference, as long as it results is on order of hours or more. Otherwise, it won't add enough randomness.

Reply:
You will get rid of the "reset time of the day" feature. Generally I don't like refactoring commits to also change behavior.

Reply:
Okay, made it seconds again so that we don't change the behavior. Although I think we won't loose any properties if hours are used.